### PR TITLE
refactor: Remove early row count and update batch export metrics

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

Counting rows is quite expensive. Let's not do it.

## Changes

* Do not count rows when starting a batch export, if a run has no rows, query will not return any records.
* Early return moved to `insert_*` batch export activities: If we peek into the record generator and we don't get anything, then we return early.
  * Added tests for this.
* App metrics no longer relies on `records_total_count` but now queries clickhouse to get a count of rows. This could potentially be very expensive.
  * IMO a metric based on number of events should never exist. We should instead look at count of runs only. 

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?
Updated a bunch of tests + added new tests that run workflows without events.

### CODE REVIEW

File: posthog/batch_exports/models.py [hunk 1]
```diff
@@ -116,6 +117,45 @@ class Status(models.TextChoices):
...
+
+    if include_events:
+        include_events_statement = f"AND event IN ({','.join(include_events)})"
+    else:
+        include_events_statement = ""
+
+    data_interval_start_ch = data_interval_start.strftime("%Y-%m-%d %H:%M:%S")
+    data_interval_end_ch = data_interval_end.strftime("%Y-%m-%d %H:%M:%S")
+
+    clickhouse_query = f"""
```
[the remark this answers is not in the document]
Comment (tomasfarias) [reply posthog/batch_exports/models.py:142]: It isn't, `include_events` and `exclude_events` are `join`-ed in the query. But I 100% agree that's an ugly way of doing it and should instead pass it as a parameter. I'll update this.

[the remark this answers is not in the document]
Comment (tomasfarias) [reply posthog/batch_exports/models.py:142]: I understood what you meant, you are correct. I've fixed both things: Both what you mentioned and what I thought you mentioned :sweat_smile: Overall everything should be correct and look better! Thank you.

File: posthog/api/app_metrics.py [hunk 2]
```diff
@@ -120,22 +94,52 @@ def get_batch_export_runs_app_metrics_queryset(self, batch_export_id: str):
...
-            BatchExportRun.objects.filter(batch_export_id=batch_export_uuid, last_updated_at__range=date_range)
-            .annotate(dates=TruncDay("last_updated_at"))
-            .values("dates")
-            .annotate(
-                successes=Sum(
-                    Coalesce("records_total_count", 0), filter=Q(status=BatchExportRun.Status.COMPLETED), default=0
-                ),
-                failures=Sum(
-                    Coalesce("records_total_count", 0), filter=~Q(status=BatchExportRun.Status.COMPLETED), default=0
+        runs = (
```
[the remark this answers is not in the document]
Comment (tomasfarias) [reply posthog/api/app_metrics.py:97]: What if nothing failed but it was retried (manually, by the user) anyways? I think it makes sense to show duplicates if the user requests duplicates. So, we need all runs.

[the remark this answers is not in the document]
Comment (tomasfarias) [reply posthog/api/app_metrics.py:97]: Moreover, no new runs are created in the event of failure, unless manually requested by us  or users. Retries are part of a run, not a separate run.

[the remark this answers is not in the document]
Comment (tiina303) [reply posthog/api/app_metrics.py:97]: 👍  would you mind changing the labels in the UI too with this PR 'events' -> 'runs' in that case?

[the remark this answers is not in the document]
Comment (tomasfarias) [reply posthog/api/app_metrics.py:96]: Sure, latest commit changed this to runs. I'll make the UI change next.

[the remark this answers is not in the document]
Comment (tomasfarias) [reply posthog/api/app_metrics.py:96]: Done in latest commit, looks like this: 
image

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**